extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -479,31 +474,81 @@ export class AmpAnalytics extends AMP.BaseElement {
     // Precedence is event.vars > trigger.vars > config.vars.
     // Nested expansion not supported.
     return expandTemplate(template, key => {
-      const match = key.match(/([^(]*)(\([^)]*\))?/);
-      const name = match[1];
-      const argList = match[2] || '';
+      const [name, argList] = this.getNameArgs_(key);
       let raw = (event && event['vars'] && event['vars'][name]) ||
           (trigger['vars'] && trigger['vars'][name]) ||
           (this.config_['vars'] && this.config_['vars'][name]) ||
           '';
+
+      // Values can also be arrays and objects. Don't expand them.
       if (typeof raw == 'string') {
         raw = this.expandTemplate_(raw, trigger, event, opt_iterations - 1);
       }
       const val = opt_encode ? this.encodeVars_(raw, name) : raw;
-      return val + argList;
+      return val ? val + argList : val;
     });
   }
 
   /**
+   * Returns an array containing two values: name and args parsed from the key.
+   *
+   * @param {string} key The key to be parsed.
+   * @return {Array<string>}
+   * @private
+   */
+  getNameArgs_(key) {
+    if (!key) {
+      return ['', ''];
+    }
+    const match = key.match(/([^(]*)(\([^)]*\))?/);
+    return [match[1], match[2] || ''];
+  }
+
+  /**
    * @param {string} raw The values to URI encode.
    * @param {string} unusedName Name of the variable.
+   * @return {string} The encoded value.
    * @private
    */
   encodeVars_(raw, unusedName) {
+    if (!raw) {
+      return '';
+    }
+
     if (isArray(raw)) {
       return raw.map(encodeURIComponent).join(',');
     }
-    return encodeURIComponent(raw);
+    // Separate out names and arguments from the value and encode the value.
+    const [valName, valArgsList] = this.getNameArgs_(String(raw));
```

Comment:
We 100% cannot do this. This will add a _massive_ Symbol polyfill into the codebase. You can use an object pattern instead:

```js
const { valName, valArgsList } = this.getNameArgs_(); // #getNameArgs should return an object, not an array.
```